File: app/src/main/java/tool/xfy9326/naucourse/fragments/UpdateSettingsFragment.java
package tool.xfy9326.naucourse.fragments;

import android.os.Bundle;

import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import java.util.Objects;

import tool.xfy9326.naucourse.Config;
import tool.xfy9326.naucourse.R;
import tool.xfy9326.naucourse.methods.UpdateMethod;

public class UpdateSettingsFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        addPreferencesFromResource(R.xml.settings_update);
    }

    @Override
    public void onResume() {
        super.onResume();
        PreferenceSet();
    }

    @SuppressWarnings("SameReturnValue")
    private void PreferenceSet() {
        ((Preference) Objects.requireNonNull(findPreference(Config.PREFERENCE_UPDATE_NOW))).setOnPreferenceClickListener(preference -> {
            if (isAdded()) {
                UpdateMethod.checkUpdate(getActivity(), true);
            }
            return false;
        });
    }
}

File: app/src/main/java/tool/xfy9326/naucourse/ui/fragments/settings/UpdateSettingsFragment.kt
package tool.xfy9326.naucourse.ui.fragments.settings

import android.os.Bundle
import androidx.lifecycle.lifecycleScope
import androidx.preference.Preference
import kotlinx.coroutines.launch
import tool.xfy9326.naucourse.R
import tool.xfy9326.naucourse.constants.PrefConst
import tool.xfy9326.naucourse.kt.showShortToast
import tool.xfy9326.naucourse.ui.dialogs.UpdateDialog
import tool.xfy9326.naucourse.ui.fragments.base.BaseSettingsPreferenceFragment
import tool.xfy9326.naucourse.update.UpdateChecker

@Suppress("unused")
class UpdateSettingsFragment : BaseSettingsPreferenceFragment() {
    override val preferenceResId = R.xml.settings_update
    override val titleName: Int = R.string.settings_update

    override fun onPrefViewInit(savedInstanceState: Bundle?) {
        findPreference<Preference>(PrefConst.CheckUpdatesNow)?.setOnPreferenceClickListener {
            checkUpdates()
            true
        }
    }

    private fun checkUpdates() {
        showShortToast(R.string.checking_update)
        lifecycleScope.launch {
            val updateInfo = UpdateChecker.getNewUpdateInfo(true)
            if (updateInfo == null) {
                showShortToast(R.string.update_check_failed)
            } else {
                if (updateInfo.first) {
                    UpdateDialog.showDialog(parentFragmentManager, updateInfo.second!!)
                } else {
                    showShortToast(R.string.no_new_update)
                }
            }
        }
    }
}